File: leetcode/algorithms/maximum_subarray/maximum_subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        total_max = current_max = nums[0]

        for value in nums[1:]:
            current_max = max(value, value + current_max)
            total_max = max(current_max, total_max)

        return total_max

File: leetcode/algorithms/maximum_subarray/maximum_subarray.cc
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int total_max = nums[0];
        int current_max = nums[0];

        for (int i = 1; i < nums.size(); ++i) {
            current_max = max(nums[i], nums[i] + current_max);
            total_max = max(current_max, total_max);
        }

        return total_max;
    }
};
